python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -217,24 +107,51 @@ class IdentityPartitionMapping(PartitionMapping, NamedTuple("_IdentityPartitionM
     partitions in the downstream asset to the same partition in the upstream asset.
     """
 
-    def get_upstream_partitions_for_partition_range(
+    def get_upstream_mapped_partitions_result_for_partitions(
         self,
-        downstream_partition_key_range: Optional[PartitionKeyRange],
-        downstream_partitions_def: Optional[PartitionsDefinition],
+        downstream_partitions_subset: Optional[PartitionsSubset],
         upstream_partitions_def: PartitionsDefinition,
-    ) -> PartitionKeyRange:
-        if downstream_partitions_def is None or downstream_partition_key_range is None:
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> UpstreamPartitionsResult:
+        if downstream_partitions_subset is None:
             check.failed("downstream asset is not partitioned")
 
-        return downstream_partition_key_range
+        if downstream_partitions_subset.partitions_def == upstream_partitions_def:
+            return UpstreamPartitionsResult(
+                downstream_partitions_subset,
+                [],
+            )
+
+        upstream_partition_keys = set(upstream_partitions_def.get_partition_keys())
```

Comment:
Do we need to be passing in the dynamic partitions store?

Reply:
No, dynamic partitions use the regular default partitions subset which serializes the partition keys in the subset. We only need to pass in the dynamic partitions store for operations that require fetching partition keys outside of the set (i.e. `get_partition_keys_not_in_subset`)

Reply:
What about for `PartitionsDefinition.get_partition_keys` though, which we're calling for the upstream?

Reply:
oooo good catch, yes that should be there